### PR TITLE
refactor: Separate the Model module into 2 submodules

### LINKED ISSUE #63: [4.2.8] Réduction de la dette technique
as:
Non pertinent

i want:
Modifier le code actuel pour faciliter le développement actuel et futur.

description:
Préparation de l'architecture du repo pour le multi-unités contre multi-unités et amélioration de l'arbre des modules.

definition of done:
Simplification des structures
Révision de l'architecture du code
Création de nouveaux modules pour mieux séparer les fonctions

workload: 3

### PR DESCRIPTION
**Pull request by Pepiloto**
Closes #63

### CODE REVIEW

**Review by TheVengeur — DISMISSED** [review]

**Review by H4ss — APPROVED** [review]
Structurally it makes sense, I have no logic problem to report.

**Review by TheVengeur — APPROVED** [review]

**Review by Jordan-B1 — COMMENTED** [review]
Does the norm commit changed ?
You should tell us when you do this kind of changes

File: src/fight/computation_tools.rs [hunk 1]
```diff
@@ -45,17 +46,14 @@ pub fn compute_roll_to_wound(strength: usize, resilience: usize) -> usize {
 
 /// Compute the probability for a model to wound an another
 ///
-/// ## Parameters
-/// (&model::Stats) attacking_stats: The attacker stats
+/// # Parameters
+/// attacking_stats (&model::Stats): The attacker stats
```
[the remark this answers is not in the document]
Comment (Pepiloto) [reply src/fight/computation_tools.rs:50]: They were 3 types of documentation in the code existing at the same time, I choose the one I thought were the one we chose before but I may have made an error.
I think it doesn't really matter which norm is it for the documentation if it's always the same.
Is it fine if we change to this format ?

[the remark this answers is not in the document]
Comment (Jordan-B1) [reply src/fight/computation_tools.rs:50]: good to me
Just post on the Dicsord server the new format so it will be clear for everyone

**Review by Jordan-B1 — APPROVED** [review]

**Review by TheVengeur — APPROVED** [review]

**Review by H4ss — APPROVED** [review]

**Pull request merged**